pym/portage/package/ebuild/config.py

```diff
@@ -2729,6 +2729,16 @@ def environ(self):
 		if not eapi_exports_merge_type(eapi):
 			mydict.pop("MERGE_TYPE", None)
 
+		if eapi_attrs.sysroot and _phase_func_map.get(phase, '').startswith('src_'):
+			# TODO: implement Portage-wide support for custom SYSROOT
+			mydict["SYSROOT"] = (
+					self.get('SYSROOT', '').rstrip(os.path.sep) + os.path.sep)
+			mydict["ESYSROOT"] = (
+					mydict["SYSROOT"].rstrip(os.path.sep) + self.get("EPREFIX", "")
+					+ os.path.sep)
+		if eapi_attrs.broot and _phase_func_map.get(phase, '').startswith('src_'):
+			mydict["BROOT"] = self.get("EPREFIX", "") + os.path.sep
```

Comment:
It seems like BROOT should begin with the value of `${ROOT}` here, since we join EPREFIX with ROOT for cross-prefix support.

Reply:
No, BROOT is supposed to refer to /, not ROOT.

Reply:
Okay, then we should use portage.const.EPREFIX (actually better to refer to it indirectly as PORTAGE_OVERRIDE_EPREFIX) here.

Reply:
I'd like to eliminate `portage.const.EPREFIX`, see https://bugs.gentoo.org/643920, so I prefer indirect reference via `self['PORTAGE_OVERRIDE_EPREFIX']`.